Marlin/src/gcode/sd/M21_M22.cpp

```diff
@@ -38,7 +42,11 @@ void GcodeSuite::M21() { card.mount(); }
  */
 void GcodeSuite::M22() {
   if (!IS_SD_PRINTING()) card.release();
-  IF_ENABLED(TFT_COLOR_UI, ui.refresh(LCDVIEW_CALL_REDRAW_NEXT));
+  #if HAS_LCD_MENU && ENABLED(TFT_COLOR_UI)
+    // Menu display issue on item removal with multi language selection menu
+    if (encoderTopLine > 0) encoderTopLine--;
+    ui.refresh(LCDVIEW_CALL_REDRAW_NEXT);
```

Comment:
also tried a simple ui.refresh(); like in card.mount()... same behavior

Reply:
A change to `encoderTopLine` should only be needed if `encoderTopLine` is also the last item in the menu, but the menu would have to have only one item in it for that to happen. If an item is removed and the currently-selected line (`encoderLine`) is the last line, then `encoderLine--` will help. Of course, this value comes from the encoder position, so `ui.encoderPosition` also needs to be modified for that case. As in `scroll_screen`…
```cpp
ui.encoderPosition = encoderLine * (ENCODER_STEPS_PER_MENU_ITEM);
```

Reply:
…That said, the menu code should already have handling for the case of encoder position being past the last item.

Reply:
yea its just the top line which is moved up, to keep the last item on the bottom, in this case the menu is correctly refreshed.

else HAS_LCD_MENU && ENABLED(TFT_COLOR_UI) may be useless, i think TFT_COLOR_UI includes HAS_LCD_MENU

Reply:
What happens if there is no previous menu item available to set as the top line – as is the case when the menu has exactly the number of items that fills the screen? The menu should become shorter, with the last line erased. Perhaps the menu code should simply be altered to erase all lines below the last menu item.

Reply:
i think it does in most cases, when you go back from a submenu there is no refresh issue like that.

sorry i left this problem on hold to work/debug the pwm "issues"

Reply:
the code is there because the menu item just call a gcode command... i remember we tried to edit menus to add the refresh previously...

Reply:
> when you go back from a submenu…

…the screen is erased before redrawing. `ui.refresh` should also trigger an erase-and-redraw.

Reply:
should, but doesnt do, i tried too, tried ui.refresh() alone and some other params. and without, its worst... no refresh :p